### PR TITLE
[PAPI-289] Friendly error messages cross-OS

### PR DESCRIPTION
**Pull request by drmathias**

### CODE REVIEW

File: src/Opdex.Platform.Infrastructure/Clients/CirrusFullNodeApi/Extensions/TransactionErrors.cs [hunk 1]
```diff
@@ -50,20 +50,21 @@ private static class General
...
+            return true;
         }
 
         /// <summary>
         /// Attempts to match an overflow exception message.
         /// </summary>
         private static bool IsOverflowException(string input)
         {
-            return input.StartsWith(typeof(OverflowException).FullName);
+            return input.StartsWith(typeof(OverflowException).FullName!);
```
Comment (mrtpain) [inline src/Opdex.Platform.Infrastructure/Clients/CirrusFullNodeApi/Extensions/TransactionErrors.cs:67]: What does the `!` at the end do?

File: src/Opdex.Platform.Infrastructure/Clients/CirrusFullNodeApi/Extensions/TransactionErrors.cs [hunk 2]
```diff
@@ -218,7 +219,7 @@ public static class Opdex
 
 
         // compiled Regex is much faster, should be initialize on startup
-        private static readonly Regex OpdexErrorRegex = new(@"(?<=OPDEX:\s).+?(?=\r\n)", RegexOptions.Compiled, TimeSpan.FromMilliseconds(500));
+        private static readonly Regex OpdexErrorRegex = new(@$"(?<=OPDEX:\s).+?(?={Environment.NewLine})", RegexOptions.Compiled, TimeSpan.FromMilliseconds(500));
```
Comment (mrtpain) [inline src/Opdex.Platform.Infrastructure/Clients/CirrusFullNodeApi/Extensions/TransactionErrors.cs:222]: Is this a donet thing? If so, thats pretty handy.

**Review by mrtpain — APPROVED** [review]
Just the one question about the `!` for my own knowledge, otherwise lgtm

File: src/Opdex.Platform.Infrastructure/Clients/CirrusFullNodeApi/Extensions/TransactionErrors.cs (hunk 1 above)
Comment (drmathias) [reply src/Opdex.Platform.Infrastructure/Clients/CirrusFullNodeApi/Extensions/TransactionErrors.cs:67]: `Type.FullName` is nullable, the ! indicates to analyzers and the compiler that it is never going to be null.

**Pull request merged**